tt_metal/impl/dispatch/program_command_sequence.hpp

```diff
@@ -22,7 +22,9 @@ class CircularBuffer;
 
 struct ProgramCommandSequence {
     HostMemDeviceCommand preamble_command_sequence;
-    HostMemDeviceCommand stall_command_sequence;
+    HostMemDeviceCommand* stall_command_sequence_ptr;
+    HostMemDeviceCommand uncached_bin_stall_command_sequence;
+    HostMemDeviceCommand cached_bin_stall_command_sequence;
```

Comment:
do you really need all three?

Reply:
yes, host needs to issue a different stall sequence during `enqueue_program` depending on whether the program was cached on device or not. The `stall_command_sequence_ptr` will store either `uncached_bin_stall_command_sequence` or `cached_bin_stall_command_sequence` depending on the state of kernel binaries in DRAM.

Reply:
@cfjchu  @tt-asaigal  I'd actually rewrite it a little bit to avoid future confusions.
Instead of 
```
HostMemDeviceCommand uncached_bin_stall_command_sequence;
HostMemDeviceCommand cached_bin_stall_command_sequence;
```
you create HostMemDeviceCommand tall_command_sequences[2];  instead of HostMemDeviceCommand* stall_command_sequence_ptr; you just set size_t current_stall_cq_seq = 0; you can even create two consts kCachedIdx = 0; and KUnchachedIdx = 1;  wdyt?

Reply:
@dmakoviichuk-tt I've updated this based on the your suggestion